### DISCUSSION TITLE
How would I go about putting a GIF on top of a Still Image?

### DISCUSSION THREAD

**Discussion by QtKaii** [SixLabors/ImageSharp#1450] [Q&A]
I am trying to produce a profile system that generates a card for each user, A user can use a GIF as their profile picture. I know drawimage only does one frame. So how would I go about having the still image in the background for all frames and having the GIF in the foreground?

**Comment by antonfirsov** [comment]
You can iterate through all frames of the overlay gif (`overlayGifImage.Frames`) and do the following:
1.  Use [`ExportFrame`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.ImageFrameCollection.html#SixLabors_ImageSharp_ImageFrameCollection_ExportFrame_System_Int32_) (or the [generic variant](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.ImageFrameCollection-1.html#SixLabors_ImageSharp_ImageFrameCollection_1_ExportFrame_System_Int32_)) to grab it as an `Image` (or `Image<T>`)
2. Draw the extracted image on the the source image: `var tempImage = sourceImage.Clone(x => x.DrawImage(...))`.
3. Add the resulting `tempImage` as a frame to your result: `resultImage.Frames.AddFrame(tempImage.RootFrame)`

@JimBobSquarePants wondering if it would be "more correct" behavior if `DrawImageProcessor` would do this automatically for multi-frame overlay images?

**Reply by JimBobSquarePants** [reply]
@antonfirsov  Are you suggesting that if you have a single frame image then try to draw a multi-frame image on top of it that we should automatically generate frames in the destination to match the source count?

If so, that's an interesting proposal that I think is worth consideration..

**Reply by antonfirsov** [reply]
Yes, however unsure if it makes sense to be this smart, or at least it means we need to clarify it further:
- What about other cases when user draws `N` frames on top of `M` frames? `(1 < N < M`, `1 < M < N`)
- What about other processors, and ImageBrush in Drawing?

Also: it would be a breaking change.
